### DISCUSSION TITLE
Wish list for new nodes

### DISCUSSION THREAD

**Discussion by Suzie1** [WASasquatch/was-node-suite-comfyui#78] [Ideas]
Hi, Fantastic custom nodes! Thank you so much for all your hard work.

Here is my wish list for new custom nodes:

1. Input switches with widget for switching (e.g. select 1 or 2)
2. Input switches for Model and CLIP
3. Latent Output switch - use case for Hires fix switch between latent upscale or image upscale
4. Load Lora with built in On/Off switch - this would remove the need for switching around the nodes which can get very messy
5. On/Off switch to switch off downstream processing - use case is for testing large worflows, or as an alternative to Latent Output Switch
6. Noise nodes for White noise and Pink noise - https://en.wikipedia.org/wiki/Colors_of_noise

Suzie
https://civitai.com/user/Akatsuzi/

**Comment by WASasquatch** [comment]
We don't (still) have a simple on/off toggle. Gotta use lists as true/false currently. I don't think I can make a basic widget toggle as that would be a front end extension. The rest seem doable besides downstream on/off. I haven't figured out how to do that without comfy throwing an error where other nodes expect a input. Error won't break anything, but not pretty either. Haha

**Comment by RockOfFire** [comment]
For the widget, I did this: "Input": ("INT", {"default": 1, "min": 1, "max": 2}), for the input and ComfyUI is able to toggle between widget and input.  I made one after you did because I had an old version of your WAS suite and didn't realize you made one aleady.

The WAS Suite is an amazing tool and you've done so much with it!  I'm really impressed with all the nodes you've made.

**Comment by RockOfFire** [comment]
Oh sorry, here's how I made my node if you want to reference it:

class ComfyRoll_InputImages:
    def __init__(self):
        pass

    @classmethod
    def INPUT_TYPES(cls):
        return {
            "required": {
                "Input": ("INT", {"default": 1, "min": 1, "max": 2}),
                "image1": ("IMAGE",),
                "image2": ("IMAGE",)
            }
        }

    RETURN_TYPES = ("IMAGE",)
    OUTPUT_NODE = True
    FUNCTION = "InputImages"

    CATEGORY = "Comfyroll"

    def InputImages(self, Input, image1, image2):
        if Input == 1:
            return (image1, )
        else:
            return (image2, )

**Comment by Suzie1** [comment]
Hi, that sounds awesome!

RockOfFire was helping me with switch nodes before we saw your latest update. He also has some code for a Load LoRA node incorporating a switch . I'll ask him to send it.

Suzie

**Comment by WASasquatch** [comment]
I prefer to have a number input for logic, because the base use is logic with dynamic workflows. Something causes the logic boolean to be either 1, or 0, and switches all the inputs. 

![image](https://github.com/WASasquatch/was-node-suite-comfyui/assets/1151589/4e1686b1-bd9c-4a2b-8505-05d96e7aa008)

**Comment by WASasquatch** [comment]
Also added a none to lora:

![image](https://github.com/WASasquatch/was-node-suite-comfyui/assets/1151589/b43d365a-2609-4be7-bc56-44349fb4721b)

**Comment by RockOfFire** [comment]
That makes sense and I really like your implementation of the Load Lora node!!  That's really cool!!  :D

**Comment by Suzie1** [comment]
Yes that will be perfect, thank you! I'll update my templates with the new nodes when they come out.

**Comment by WASasquatch** [comment]
I pushed the update awhile ago so should be able to pull.

**Comment by WASasquatch** [comment]
@Suzie1 I've added a Image Power Noise node:

![image](https://github.com/WASasquatch/was-node-suite-comfyui/assets/1151589/89d36fd1-f856-4998-a6c4-2dd3d34a7934)

**Comment by RockOfFire** [comment]
That looks so cool!!  :D   Those Image Power Noise nodes are amazing!!

**Comment by Suzie1** [comment]
Yes, thank you, these will be perfect!

**Comment by curiousjp** [comment]
Hello - I think it would be neat to have a node that takes a list and text and tests to see if the first includes the second, outputting a boolean number that can be used by things like WAS_VAE_Input_Switch - combined with the name_string return from the checkpoint loader, these can be used to e.g. build a whitelist of checkpoints that can use their own baked VAE.

I have actually written something along these lines, but I'm not exactly sure how to use git to make a pull request. If it's useful though, it would be something like this -
```Python
class WAS_Text_List_Contains:
    def __init__(self):
        pass
    @classmethod
    def INPUT_TYPES(cls):
        return {
            "required": {
                "list": ("LIST", {"forceInput": True}),
                "text": ("STRING", {"forceInput": True}),
            },
        }
    RETURN_TYPES = ("NUMBER",)
    FUNCTION = "list_contains"
    CATEGORY = "WAS Suite/Text"
    def list_contains(self, list, text):
        if text in list:
            return (1,)
        else:
            return (0,)
        
```
Thank you for your hard work on these extensions to Comfy. They have let me port through a lot of my existing workflows in ways that will give me significant room for expanding them in the future.  (My apologies in advance if this node already exists and I just can't see it for some reason.)

**Reply by WASasquatch** [reply]
Git is a little intense in that aspect. What I do honestly is just fork the repo, edit it directly, then at the home of the repo it say it's X commits ahead of WAS-NS and you can click there to make a PR
